daprdocs/content/en/operations/components/pluggable-components/pluggable-components-discovery.md

```diff
@@ -0,0 +1,229 @@
+linkTitle: "How To: Discover a pluggable component"
+weight: 4500
+description: "Learn how to help Dapr discover your pluggable component"
+---
+
+[uds]: https://en.wikipedia.org/wiki/Unix_domain_socket
+
+## Service Discovery Process
+
+Pluggable, [gRPC-based](https://grpc.io/) components are typically run as containers or processes that need to communicate with the Dapr main process via [Unix Domain Sockets][uds]. They are automatically discovered and registered in runtime by Dapr using the following steps:
+
+1. The Component listens to an [Unix Domain Socket][uds] placed on the shared volume.
+2. The Dapr runtime lists all [Unix Domain Socket][uds] in the shared volume.
+3. The Dapr runtime connects with the socket and uses gRPC reflection to discover all services that such component implements.
+
+A single component can implement multiple [building blocks]({{< ref building-blocks-concept.md >}}) at once.
+
+<img src="/images/grpc-components.png" width=50%>
+
+While Dapr's built-in components come [ready to be used out of the box](https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md), pluggable components require a few setup steps before they can be used with Dapr.
+
+1. Pluggable components need to be started and ready to take requests _before_ Dapr itself is started.
+2. The [Unix Domain Socket][uds] file used used for the pluggable component communication need to be made accessible to both Dapr and pluggable component.
+
+Dapr does not interfere with orchestrating components containers creation and deployment. This is your domain, and it will be different depending on how Dapr and your components are run:
+
+- In standalone mode, as processes or containers, or
+- In Kubernetes, as containers.
+
+This will also change the mechanisms available to share [Unix Domain Socket][uds] files between Dapr and pluggable components.
+
+{{% alert title="Note" color="primary" %}}
+As a prerequisite the running operating system must supports Unix Domain Sockets, any UNIX or UNIX-like system (Mac, Linux, or for local development [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) for Windows users) should be sufficient.
+{{% /alert %}}
+
+Select your running environment to begin making your component discoverable by Dapr.
+
+{{< tabs "Standalone" "Kubernetes" >}}
+
+{{% codetab %}}
+[uds]: https://en.wikipedia.org/wiki/Unix_domain_socket
+
+## Run the component
+
+As mentioned previously, your component and the Unix Socket must be up and running before Dapr starts.
+
+By default, Dapr looks for [Unix Domain Socket][uds] files in the folder in `/tmp/dapr-components-sockets`.
+
+The name of the file without any extension will be the name of the component. For example, for `memstore.sock`, the component name will be `memstore`.
+
+Since you are running Dapr in the same host as the component, simply verify this folder and the files within it are accessible and writable by both your component and Dapr.
```

Comment:
```suggestion
Since you are running Dapr in the same host as the component, verify this folder and the files within it are accessible and writable by both your component and Dapr.
```